src/Controller/ProfileController.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace App\Controller;
+
+use App\Entity\User;
+use App\Form\PasswordChangeType;
+use App\Service\PasswordManager;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\PasswordHasher\Hasher\UserPasswordHasherInterface;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Security\Core\Encoder\EncoderFactoryInterface;
+
+/**
+ * @Route("/profile", name="profile_")
+ */
+class ProfileController extends AbstractController
+{
+    /**
+     * @Route("/password", name="password")
+     */
+    public function changePassword(
+        Request $request,
+        UserPasswordHasherInterface $userPasswordHasher,
+        EntityManagerInterface $entityManager,
+        PasswordManager $passwordManager
+    ): Response {
+        $isUser = $this->getUser();
+
+        if ($isUser && ($isUser instanceof User)) {
+            $user = $this->getDoctrine()
+                ->getRepository(User::class)
+                ->findOneBy(['email' => $isUser->getEmail()]);
```

Comment:
$this->getUser() already gives you a User, so why do you call a findOneBy to get the same User ?
And also $this->getDoctrine()->getRepository(...) <== This is not a good way to get a repository, use the service injection instead